### DISCUSSION TITLE
Measuring end-to-end latency

### DISCUSSION THREAD

**Discussion by jonathan-thorpe** [savonet/liquidsoap#3779] [General]
Hi All,

I have two LiquidSoap instances forming a studio-to-transmitter link that services multiple sites. It’s important that I try to maintain these reasonably in sync in terms of latency.

Does anyone have any creative ideas on how I can inject a precise time source at the source and then compare this at the sink to report how much actual latency in ms I might have?

**Comment by toots** [comment]
I'm not sure how to measure it but I think that:
* `srt` should be the output of choice for that. Might want to consider a raw format as well.
* You can reduce the size of the internal frame following https://www.liquidsoap.info/doc-2.2.4/cookbook.html#alsa-unbuffered-output , in `main`/`2.3.x`, this is supported out of the box now.
* SRT can report the delay via the stats I believe.

This should help you compute and upper bound for the expected latency. I think it would be: input frame size + srt latency + output frame size
